contracts/modules/tokenswap/TokenSwapModule.sol

```diff
@@ -344,46 +433,144 @@ contract TokenSwapModule is ModuleBaseWithFee {
         uint256 tokenArrayLength = _ts.tokens.length;
         for (uint256 i; i < tokenArrayLength; ++i) {
             uint256[] memory pt = _ts.pathTo[i];
-            address token = _ts.tokens[i];
             uint256 pathArrayLength = pt.length >> 2;
+            uint256 totalRewardForToken = 0;
+            uint256 reward = 0;
+            uint256 amount = 0;
             for (uint256 k; k < pathArrayLength; ++k) {
-                // every 4 values, the values for a new dao start
-                // value 0 = instant amount
-                // value 1 = vested amount
-                // value 2 = vesting cliff
-                // value 3 = vesting duration
-                uint256 instant = pt[k << 2];
-                uint256 vested = pt[(k << 2) + 1];
-
-                if (instant > 0) {
-                    amountsOut[i] += instant;
-                    _transferWithFee(token, _ts.daos[k], instant);
-                }
-
-                if (vested > 0) {
-                    amountsOut[i] += vested;
-                    uint256 amount = _payFeeAndReturnRemainder(token, vested);
-                    address daoDepositManager = dealManager
-                        .getDaoDepositManager(_ts.daos[k]);
-                    if (token != address(0)) {
-                        _approveDaoDepositManager(token, _ts.daos[k], amount);
-                    }
-
-                    uint256 callValue = token == address(0) ? amount : 0;
-                    IDaoDepositManager(daoDepositManager).startVesting{
-                        value: callValue
-                    }(
-                        _dealId,
-                        token,
-                        amount, // amount
-                        uint32(pt[(k << 2) + 2]), // cliff
-                        uint32(pt[(k << 2) + 3]) // duration
-                    );
-                }
+                // Distributes the instant and vested amount for the given token
+                (amount, reward) = _distributeInstantAndVestedToken(
```

Comment:
Maybe the function could be renamed to something simpler, like `_distributeToken` or so? The instant + vested part is basically all options, so not sure if there is a big value in writing it out? You can decide that of course, just to me the initial reaction was, that there must be something else compared to instant and vested. Maybe you thought about the reward receivers? In that case, I can understand it, maybe something like `_distributeTokenToDAO`?

Edit: I see that you now have `_distributeInstant..` and `_distributeVested...`, so it makes a bit more sense since this one combines it. Personally I am still leaning towards the `_distributeTokenToDAO` approach I think. You decide :)

Reply:
makes sense, changed it